CONTRIBUTING.md

```diff
@@ -58,19 +58,130 @@ can be accepted.
 
 - Each new feature or change PR should provide meaningful CI tests.
 - All the tests and security scanning checks in CI should pass and achieve the expected coverage.
+- To avoid unnecessary failures in GitHub Actions, make sure `make check` and `make test` work locally.
+See below for instructions to set up the development environment.
 
 ### Merging PRs
 
-- Before a PR is merged, all commits in the PR should be rebased into meaningful commits.
+- Before a PR is merged, all commits in the PR should meaningful commits.
```

Comment:
```suggestion
- Before a PR is merged, all commits in the PR should have meaningful commits.
```
We could use another word if it make more sense.

Reply:
See commit [6de60e8](https://github.com/oracle-samples/macaron/pull/279/commits/6de60e887c592f5ef3b4bb984985148507222b1f)